core/server/common/src/main/java/alluxio/master/journal/raft/RaftJournalWriter.java

```diff
@@ -82,16 +85,23 @@ public void write(JournalEntry entry) throws IOException, JournalClosedException
     }
     Preconditions.checkState(entry.getAllFields().size() <= 1,
         "Raft journal entries should never set multiple fields, but found %s", entry);
+    if (mCurrentJournalEntrySize.get() > mFlushBatchBytes) {
+      flush();
+    }
     if (mJournalEntryBuilder == null) {
       mJournalEntryBuilder = JournalEntry.newBuilder();
       mCurrentJournalEntrySize.set(0);
     }
     LOG.trace("Writing entry {}: {}", mNextSequenceNumberToWrite, entry);
     mJournalEntryBuilder.addJournalEntries(entry.toBuilder()
         .setSequenceNumber(mNextSequenceNumberToWrite.getAndIncrement()).build());
-    if (mCurrentJournalEntrySize.addAndGet(entry.getSerializedSize()) > mFlushBatchBytes) {
-      flush();
+    long size = entry.getSerializedSize();
+    if (size > mEntrySizeMax) {
+      LOG.error("Journal entry size ({}) is bigger than the max allowed size ({}) defined by {}",
```

Comment:
This is confusing, since the max entry size is really `MASTER_EMBEDDED_JOURNAL_ENTRY_SIZE_MAX`, but `mEntrySizeMax` will be 1/3 of that. So, this message will say something like `entry size (150) is bigger than max (100) defined by alluxio.master.embedded.journal.entry.size.max` However, `alluxio.master.embedded.journal.entry.size.max` will be set to 300.

Reply:
```
    mEntrySizeMax = ServerConfiguration
        .getBytes(PropertyKey.MASTER_EMBEDDED_JOURNAL_ENTRY_SIZE_MAX);
    mFlushBatchBytes = mEntrySizeMax / 3;
```
entry size max = 10MB
flush batch bytes = 10MB / 3 = 3.3MB
here we complain that a single entry size is bigger than the entry size max 10MB

Reply:
ahhh, I see. I think I got confused with entry size and batch size.

Is this the only place we error when the entry is too large? Or will we get another error somewhere else in ratis?

Reply:
this place just gives clue for why Ratis flush failed. The actual error is logged by https://github.com/Alluxio/alluxio/pull/12649